File: modules/angular2/src/core/pipes/invalid_pipe_argument_exception.ts
import {CONST, Type} from 'angular2/src/core/facade/lang';
import {BaseException, WrappedException} from 'angular2/src/core/facade/exceptions';

export class InvalidPipeArgumentException extends BaseException {
  constructor(type: Type, value: Object) {
    super(`Invalid argument '${value}' for pipe '${type}'`);
  }
}

File: lib/src/common/pipes/invalid_pipe_argument_exception.dart
library angular2.src.common.pipes.invalid_pipe_argument_exception;

import "package:angular2/src/facade/lang.dart" show Type;
import "package:angular2/src/facade/exceptions.dart"
    show BaseException, WrappedException;

class InvalidPipeArgumentException extends BaseException {
  InvalidPipeArgumentException(Type type, Object value)
      : super('''Invalid argument \'${ value}\' for pipe \'${ type}\'''') {
    /* super call moved to initializer */;
  }
}
